apps/cowswap-frontend/src/modules/volumeFee/state/cowswapFeeAtom.ts

```diff
@@ -1,41 +1,62 @@
 import { atom } from 'jotai'
 
-import { GNOSIS_CHAIN_STABLECOINS } from '@cowprotocol/common-const'
+import { STABLECOINS } from '@cowprotocol/common-const'
 import { getCurrencyAddress, isInjectedWidget } from '@cowprotocol/common-utils'
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 import { walletInfoAtom } from '@cowprotocol/wallet'
 
 import { derivedTradeStateAtom } from 'modules/trade'
 
+import { featureFlagsAtom } from 'common/state/featureFlagsState'
+
 import { VolumeFee } from '../types'
 
 const COWSWAP_VOLUME_FEES: Record<SupportedChainId, VolumeFee | null> = {
   [SupportedChainId.MAINNET]: null,
   [SupportedChainId.SEPOLIA]: null,
-  [SupportedChainId.ARBITRUM_ONE]: null,
-  // Only Gnosis chain
+  [SupportedChainId.ARBITRUM_ONE]: {
+    bps: 10, // 0.1%
+    recipient: '0x451100Ffc88884bde4ce87adC8bB6c7Df7fACccd', // Arb1 Protocol fee safe
+  },
   [SupportedChainId.GNOSIS_CHAIN]: {
     bps: 10, // 0.1%
     recipient: '0x6b3214fD11dc91De14718DeE98Ef59bCbFcfB432', // Gnosis Chain Protocol fee safe
   },
 }
 
 export const cowSwapFeeAtom = atom((get) => {
-  const { chainId } = get(walletInfoAtom)
+  const { chainId, account } = get(walletInfoAtom)
   const tradeState = get(derivedTradeStateAtom)
+  const featureFlags = get(featureFlagsAtom)
 
   const { inputCurrency, outputCurrency } = tradeState || {}
 
-  // No widget mode
+  // Don't use it in the widget
   if (isInjectedWidget()) return null
 
+  // Don't user it when the currencies are not set
   if (!inputCurrency || !outputCurrency) return null
 
-  const isInputTokenStable = GNOSIS_CHAIN_STABLECOINS.includes(getCurrencyAddress(inputCurrency).toLowerCase())
-  const isOutputTokenStable = GNOSIS_CHAIN_STABLECOINS.includes(getCurrencyAddress(outputCurrency).toLowerCase())
+  // Don't use it when isCowSwapFeeEnabled is not enabled
+  if (!featureFlags.isCowSwapFeeEnabled) return null
```

Comment:
We currently have a fee for Gnosis chain with no feature-flag, I think this condition should be under `chainId === SupportedChainId.ARBITRUM_ONE`

Reply:
Actually, this feature flag (isCowSwapFeeEnabled) only applies to Gnosis chain, and it this is off there will be no fee there.
Maybe it makes sense to remove it completely?

Reply:
Will remove it in another PR. See thread https://cowservices.slack.com/archives/C05N1FNP3MX/p1730296706380129